lib/charms/mongos/v0/mongos_client_interface.py

```diff
@@ -123,31 +138,3 @@ def update_connection_info(self, config: MongosConfiguration) -> None:
                 relation.id,
                 config.uri,
             )
-
-
-class MongosRequirer(Object):
-    """Manage relations between the mongos router and the application on the application side."""
-
-    def __init__(
-        self,
-        charm: CharmBase,
-        database_name: str,
-        extra_user_roles: str,
-        relation_name: str = MONGOS_RELATION_NAME,
-    ) -> None:
-        """Constructor for MongosRequirer object."""
-        self.relation_name = relation_name
-        self.charm = charm
-
-        if not database_name:
-            database_name = f"{self.charm.app}"
-
-        self.database_requires = DatabaseRequires(
-            self.charm,
-            relation_name=self.relation_name,
-            database_name=database_name,
-            extra_user_roles=extra_user_roles,
-        )
-
-        super().__init__(charm, self.relation_name)
-        # TODO Future PRs handle relation broken
```

Comment:
Because it doesn't really do anything that the `DatabaseRequires ` doesn't. It is just duplicated code.

The only thing is adds is:
```
        if not database_name:
            database_name = f"{self.charm.app}"
```
which I don't think is valuable enough to warrant keeping it.